main/system.c

```diff
@@ -84,10 +87,42 @@ static void _init_system(GlobalState * global_state, SystemModule * module)
 
     ADC_init();
 
-    // DS4432U tests
-    DS4432U_set_vcore(nvs_config_get_u16(NVS_CONFIG_ASIC_VOLTAGE, CONFIG_ASIC_VOLTAGE) / 1000.0);
-
-    EMC2101_init(nvs_config_get_u16(NVS_CONFIG_INVERT_FAN_POLARITY, 1));
+    /* perform device init based on the device name */
+    /* TODO add other platforms besides HEX */
+    switch (global_state->board_version) {
+        case 201:  /* ULTRA */
+        case 202:
+        case 203:
+        case 204:
+            // DS4432U tests
+            DS4432U_set_vcore(nvs_config_get_u16(NVS_CONFIG_ASIC_VOLTAGE, CONFIG_ASIC_VOLTAGE) / 1000.0);
+
+            EMC2101_init(nvs_config_get_u16(NVS_CONFIG_INVERT_FAN_POLARITY, 1));
+            EMC2101_set_fan_speed(1);
+            break;
+        case 302:  /* HEX */
+            // Initialize the core voltage regulator
+            TPS546_init();
+            // Fan config - Hex has two fans
+            //EMC2302_init(nvs_config_get_u16(NVS_CONFIG_INVERT_FAN_POLARITY, 1));
+            //EMC2302_set_fan_speed(0, (float) nvs_config_get_u16(NVS_CONFIG_FAN_SPEED, 100) / 100);
+            //EMC2302_set_fan_speed(1, (float) nvs_config_get_u16(NVS_CONFIG_FAN_SPEED, 100) / 100);
+            // temperature sensors - Hex has two sensors
+            if (TMP1075_installed(0)) {
+                ESP_LOGI(TAG, "Temperature sensor 0: %d", TMP1075_read_temperature(0));
+            }
+            if (TMP1075_installed(1)) {
+                ESP_LOGI(TAG, "Temperature sensor 1: %d", TMP1075_read_temperature(1));
+            }
+            break;
+        case 402: /* Supra */
+            TPS546_init();
+            EMC2101_init(nvs_config_get_u16(NVS_CONFIG_INVERT_FAN_POLARITY, 1));
```

Comment:
this is already handled in current master as Supra is a Supra (without or without TPS) so the EMC2101 is initialized correctly already